src/models/trigger/factory.ts

```diff
@@ -0,0 +1,75 @@
+import * as avs_pb from "../../../grpc_codegen/avs_pb";
+import BlockTrigger from "./block";
+import { BlockTriggerProps } from "./block";
+import CronTrigger, { CronTriggerProps } from "./cron";
+import EventTrigger, { EventTriggerProps } from "./event";
+import FixedTimeTrigger, { FixedTimeTriggerProps } from "./fixedTime";
+import Trigger, { TriggerTypes } from "./interface";
+
+import { TriggerProps } from "./interface";
+import ManualTrigger, { ManualTriggerProps } from "./manual";
+
+class TriggerFactory {
+  /**
+   * Static factory method to create Trigger instances
+   * @param props
+   * @returns
+   */
+  static create(props: TriggerProps): Trigger {
```

Comment:
do you think this just cause extra unncecessary step?

```
  trigger: TriggerFactory.create({
    name: "blockTrigger",
    type: TriggerTypes.BLOCK,
    data: { interval: 5 },
  }),
```

vs 

```
trigger: new BlockTrigger({name: 'every 5 block trigger', interval: 5})
```

no need to pass type, and wrap everything into `data`

Reply:
Due to class inheritance, all Trigger class needs to share the same variable name, which is `data`. 

Also, it’s really hard to get the typescript type definition right, with some of the input `number`, but others an `object` or `array`, so a common structure `data: object` was needed.

We could do `data: 5`, but I found `data: {interval: 5}` is more descriptive.

Reply:
no i mean just do 

```
trigger: BlockTrigger.create({name: 'every 5 block trigger', interval: 5})
```

a user know exactlt what parameter a block trigger need based on the type def of the blcok trigger.

there is no value to adding the below:

```
TriggerFactory.create({
    name: "ever 5 block",
    type: TriggerTypes.BLOCK,
    data: { interval: 5 },
  }),
```

Unless we do other conversion and dynamic calculateion, right now we literally just switch type then invoke the underlying class directly.

Reply:
Yes, we can. I chose to do this way 
```
TriggerFactory.create({
    name: "ever 5 block",
    type: TriggerTypes.BLOCK,
    data: { interval: 5 },
  }),
```

It’s a little extra code to keep a unified format for all 5 triggers.